synapsemonitor/monitor.py

```diff
@@ -113,6 +113,55 @@ def _find_modified_entities_file(syn: Synapse, syn_id: str, days: int = 1) -> li
     return []
 
 
+def _traverse(
+    syn: Synapse,
+    synid_root: str,
+    include_types: typing.List = ["folder", "file", "project"],
+) -> list:
+    """Traverse Synapse entity hierarchy to gather all descendant
+    entities of a root entity.
+    Args:
+        syn: Synapse connection
+        synid_root: Synapse ID of root entity.
+        include_types: Must be a list of entity types (ie. [“folder”,”file”])
+            which can be found here:
+            http://docs.synapse.org/rest/org/sagebionetworks/repo/model/EntityType.html
+    Returns:
+        List of descendant Synapse IDs and root Synapse ID
+    """
+
+    synid_desc = []
+
+    # full traverse depends on examining folder entities, even if not requested
+    include_types_mod = set(include_types)
+    include_types_mod.add("folder")
+    include_types_mod = list(include_types_mod)
+
+    synid_children = syn.getChildren(parent=synid_root, includeTypes=include_types_mod)
+    for synid_child in synid_children:
+        entity = syn.get(synid_child["id"], downloadFile=False)
+        entity_type = (
+            entity["concreteType"].split(".")[-1].lower().replace("entity", "")
+        )
+        if entity_type in ["folder", "project"]:
```

Comment:
After thinking about this some more and looking at the code, let's actually put some time on the calendar to discuss this briefly.

* What is the use case for returning if the folders or projects have been modified?   Since containers can't be versioned, the way to tell if a folder/project has been newly created is by comparing the createdOn and modifiedOn date.  A folder or project will have a change in modifiedOn value when:
   * annotated
   * syn.store is called on it even if there is no modifications
   * wiki page is modified
   * newly created
   * (unsure) - is a folder or project modified when a file entity is stored within it? (this could be potentially useful, but we wouldn't be able to differentiate from any of the above scenarios.

If we don't include folders or projects, it simplifies this code chunk by quite a bit.  For instance, is there significance in tracking the modifiedOn values in a project with 100 folders when what we really care about is the files in those folders?